src/main/java/org.wso2.apim.monetization/impl/StripeMonetizationImpl.java

```diff
@@ -555,13 +570,182 @@ public boolean disableMonetization(String tenantDomain, API api, Map<String, Str
             log.error(errorMessage);
             throw new MonetizationException(errorMessage, e);
         } catch (APIManagementException e) {
-            String errorMessage = "Failed to get API ID from database for : " +  api.getId().getApiName() +
+            String errorMessage = "Failed to get API ID from database for : " + api.getId().getApiName() +
                     " when getting tier to billing engine plan mapping.";
             log.error(errorMessage);
             throw new MonetizationException(errorMessage, e);
         }
     }
 
+    /**
+     * Publish monetization usage count
+     *
+     * @param lastPublishInfo Info about last published task
+     * @return true if successful, false otherwise
+     * @throws MonetizationException if the action failed
+     */
+    public boolean publishMonetizationUsageRecords(MonetizationUsagePublishInfo lastPublishInfo)
+            throws MonetizationException {
+
+        String apiName = null;
+        String apiVersion = null;
+        String tenantDomain = null;
+        int applicationId;
+        String apiProvider = null;
+        Long requestCount = 0L;
+        String lastPublishedTimeStamp;
+        Long currentTimestamp;
+        int flag = 0;
+        int counter = 0;
+        MonetizationUsagePublishInfo monetizationUsagePublishInfo = new MonetizationUsagePublishInfo();
+        JSONObject jsonObj = null;
+        APIAdmin apiAdmin = new APIAdminImpl();
+
+        DateFormat df = new SimpleDateFormat(StripeMonetizationConstants.TIME_FORMAT);
+        Date dateobj = new Date();
+        df.setTimeZone(TimeZone.getTimeZone(StripeMonetizationConstants.TIME_ZONE));
+        String currentDate = df.format(dateobj);
+        currentTimestamp = getTimestamp(currentDate);
+        try {
+            APIUsageStatisticsRestClientImpl apiUsageStatisticsRestClient = new APIUsageStatisticsRestClientImpl();
+            jsonObj = apiUsageStatisticsRestClient.getUsageCountForMonetization(
+                    monetizationUsagePublishInfo.getLastPublishTime(), currentTimestamp);
+        } catch (APIMgtUsageQueryServiceClientException e) {
+            String errorMessage = "Failed to get the API Usage count for Monetization";
+            log.error(errorMessage);
+            throw new MonetizationException(errorMessage, e);
+        }
+        log.info("Usage record publisher is running");
+        SubscriptionItem subscriptionItem = null;
+        try {
+            if (jsonObj != null) {
+                JSONArray jArray = (JSONArray) jsonObj.get(APIUsageStatisticsClientConstants.RECORDS_DELIMITER);
+                for (Object record : jArray) {
+                    JSONArray recordArray = (JSONArray) record;
+                    if (recordArray.size() == 6) {
```

Comment:
Why 6? Shall we explain this a bit?

Reply:
fixed in 49ded003cbdb2832bd0e8845951ac7af9831f3cb